src/BestCrvHitDeltaT_module.cc

```diff
@@ -0,0 +1,114 @@
+//
+// Module that finds the CrvCoincidenceCluster
+// closest in time to a KalSeed
+//
+// Original author: Andy Edmonds 2022-03-14
+//
+
+// Framework includes.
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "fhiclcpp/types/Atom.h"
+#include "fhiclcpp/types/OptionalAtom.h"
+#include "fhiclcpp/types/Table.h"
+#include "fhiclcpp/types/OptionalSequence.h"
+
+#include "TrkAna/inc/BestCrvAssns.hh"
+
+// C++ includes.
+#include <iostream>
+#include <string>
+#include <cmath>
+
+using namespace std;
+
+namespace mu2e {
+
+  class BestCrvHitDeltaT : public art::EDProducer {
+
+  public:
+    struct Config {
+      using Name=fhicl::Name;
+      using Comment=fhicl::Comment;
+
+      fhicl::Atom<art::InputTag> kalSeedTag{Name("kalSeedTag"), Comment("art::InputTag for KalSeedCollection")};
+      fhicl::Atom<art::InputTag> crvCoincidenceTag{Name("crvCoincidenceTag"), Comment("art::InputTag for CrvCoincidenceClusterCollection")};
+    };
+    typedef art::EDProducer::Table<Config> Parameters;
+
+    explicit BestCrvHitDeltaT(const Parameters& conf);
+    virtual ~BestCrvHitDeltaT() { }
+
+    void beginJob();
+    void produce(art::Event& e);
+
+  private:
+
+    // fhicl parameters
+    art::InputTag _kalSeedTag;
+    art::InputTag _crvCoincidenceTag;
+  };
+
+  BestCrvHitDeltaT::BestCrvHitDeltaT(const Parameters& conf):
+    art::EDProducer(conf),
+    _kalSeedTag(conf().kalSeedTag()),
+    _crvCoincidenceTag(conf().crvCoincidenceTag())
+  {  
+    produces<BestCrvAssns>("first");
```

Comment:
Is the intent to save the 2 'best' (smallest Dt) Assns?  In that case, why not add the DeltaT has payload to the assns, and maybe save a collection of assns, sorted to have the best first?  The # to save (or max Dt) could be a config parameter.

Reply:
Hi Dave. I played around with adding the Dt as payload to the Assns and making it more configurable but I couldn't get it to work. I think the module is OK as it is though. The only reason I added a second Assns was so that I could have an example for people to follow in the future (fcl/TrkAnaReco_multipleBestCrv.fcl).